From 50440228dea086d5561ca4aef882b57b9e110348 Mon Sep 17 00:00:00 2001
From: Mark Ryan <mark.d.ryan@intel.com>
Date: Fri, 27 Mar 2015 14:52:13 +0100
Subject: [PATCH 3/3] [OV5648] Added 5MP video resolution

It seems that this resolution must be enabled in the sensor driver
even if it is not explicitly used to record video.  There is a bug
in the current GMS camera application in which it sometimes
mistakenly uses a capture request prepared for recording video to
take a Lens Blur or a Photosphere image.  Without a 5MP video
resolution in the sensor, the resulting images are corrupt.

Change-Id: Ibc2741c6f469cd8200791087fa4b30a779066d75
Issues: GMINL-6901
Signed-off-by: Mark Ryan <mark.d.ryan@intel.com>
---
 drivers/media/i2c/atomisp/ov5648.h | 15 +++++++++++++++
 1 file changed, 15 insertions(+)

diff --git a/drivers/media/i2c/atomisp/ov5648.h b/drivers/media/i2c/atomisp/ov5648.h
index 1c6fa2f..31969c3 100644
--- a/drivers/media/i2c/atomisp/ov5648.h
+++ b/drivers/media/i2c/atomisp/ov5648.h
@@ -790,6 +790,21 @@ struct ov5648_resolution ov5648_res_still[] = {
 
 struct ov5648_resolution ov5648_res_video[] = {
 	{
+	 .desc = "ov5648_5M_15fps",
+	 .width = 2592,
+	 .height = 1944,
+	 .pix_clk_freq = 84,
+	 .fps = 15,
+	 .used = 0,
+	 .pixels_per_line = 2838,
+	 .lines_per_frame = 1984,
+	 .bin_factor_x = 1,
+	 .bin_factor_y = 1,
+	 .bin_mode = 0,
+	 .skip_frames = 3,
+	 .regs = ov5648_5M_15fps_2lanes,
+	 },
+	{
 	 .desc = "ov5648_1280x960_30fps",
 	 .width = 1280,
 	 .height = 960,
-- 
1.9.1

